presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -198,13 +198,14 @@ protected ResultSet getTables(Connection connection, Optional<String> schemaName
             throws SQLException
     {
         String sql = "" +
-                "SELECT att.attname, att.attndims " +
+                "SELECT att.attname, CASE WHEN att.attndims = 0 AND attyp.typcategory = 'A' THEN 1 ELSE att.attndims END AS attndims " +
                 "FROM pg_attribute att " +
+                "  JOIN pg_type attyp ON att.atttypid = attyp.oid" +
                 "  JOIN pg_class tbl ON tbl.oid = att.attrelid " +
                 "  JOIN pg_namespace ns ON tbl.relnamespace = ns.oid " +
                 "WHERE ns.nspname = ? " +
                 "AND tbl.relname = ? " +
-                "AND att.attndims > 0 ";
+                "AND CASE WHEN att.attndims = 0 AND attyp.typcategory = 'A' THEN 1 ELSE att.attndims END > 0 ";
```

Comment:
If the table is `create table t (a _inet);`, this result is different from original query. Is it okay?
```
 attname | attndims 
---------+----------
 a       |        1
```